app/infrastructure/external/storage/s3_storage.py

```
@@ -0,0 +1,253 @@
+                return {
+                    "size": head["ContentLength"],
+                    "content_type": head.get(
+                        "ContentType", "application/octet-stream"
+                    ),
+                    "checksum": meta.get("sha256"),
+                    "last_modified": head["LastModified"].isoformat(),
+                    "custom": meta,
+                }
+            except ClientError as e:
+                if e.response["Error"]["Code"] == "404":
+                    raise StorageNotFoundError(f"File not found: {storage_ref}") from e
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)